subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/TestReport.java

```diff
@@ -105,8 +146,13 @@ private void addTo(Object result, ConfigurableFileCollection dirs) {
      * Sets the binary test results to use to include in the report. Each entry must point to a binary test results directory generated by a {@link Test}
      * task.
      */
+    @Deprecated
     public void setTestResultDirs(Iterable<File> testResultDirs) {
-        resultDirs = getObjectFactory().fileCollection();
+        DeprecationLogger.deprecateProperty(TestReport.class, "testResultDirs").replaceWith("testResults")
+            .willBeRemovedInGradle8()
+            .withDslReference()
+            .nagUser();
+        resultDirs.setFrom(getObjectFactory().fileCollection());
```

Comment:
I think `resultsDir.setFrom(testResultsDirs)` might work here too.

Reply:
In this case it works, because the argument is an iterable of files, not potentially a Test task; therefore circumventing the legacy `reportOn` and `addTo` calls is OK.